app/models/piece.rb

```diff
@@ -93,4 +93,20 @@ def remains_on_board?(to_x:, to_y:)
       false
     end
   end
+
+  def successful_capture(to_x:, to_y:)
+    return false if obstructed_diagonally?(to_x:, to_y:)
```

Comment:
But wouldn't the capture method return `false` (aka `failed`) if any of the three `if obstructed`s return `true`?